src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1836,6 +1857,19 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
             $senderReferences[$serviceId] = new Reference($serviceId);
         }
 
+        $failureTransportsByTransportName = [];
+        foreach ($config['transports'] as $name => $transport) {
+            if ($transport['failure_transport']) {
+                if (!isset($config['transports'][$transport['failure_transport']])) {
```

Comment:
I think this should be checking `$senderReferences` instead of `$config['transports']` - see the old code.